TensorMath.lua

```diff
@@ -29,12 +29,14 @@ local unpack = unpack or table.unpack
 
 -- specific to CUDA
 local typenames = {'CudaByteTensor',
-                   'CudaCharTensor',
-                   'CudaShortTensor',
-                   'CudaIntTensor',
-                   'CudaLongTensor',
-                   'CudaTensor',
-                   'CudaDoubleTensor'}
+               'CudaCharTensor',
+               'CudaShortTensor',
+               'CudaIntTensor',
+               'CudaLongTensor',
+               'CudaTensor',
+               'CudaDoubleTensor',
+               'CudaHalfTensor'
```

Comment:
ideally this should be enabled only if we are compiling with HalfTensor. know of any way to get that flag into this Lua program?